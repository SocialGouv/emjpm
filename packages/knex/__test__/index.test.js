//

const { knex } = global;

beforeAll(async () => {
  await knex.migrate.latest();
});

test("should list all tables", async () => {
  const query = `
    SELECT table_name FROM information_schema.tables
    WHERE table_schema = current_schema() AND table_catalog = ?
  `;
  const results = await knex.raw(query, [knex.client.database()]);
  expect(results.rows.map(row => row.table_name).sort()).toMatchSnapshot();
});

// !
// ! HACK(douglasduteil): use the "all table" snapshot to get the tables name
// ! As I didn't find how to make jest "top await" for the table list to
// ! arrive, I'm using the generated snapshot list above.
// ! This is very hacky but the good part is :
// ! What ever happen in database tables will affect both tests.
// ! The tests above and below will fail as they are "naturally" linked.
// !
// ! @lionelB calls this : **snapshotception**
// !
const tables = JSON.parse(
  require("./__snapshots__/index.test.js.snap")
    ["should list all tables 1"] // the name of the snapshot above.
    .trim() // 'cause we need to this this snapshot
    .slice(6) // 'cause it's definitely an "Array ". That's makes 6 characters
    .replace(",\n]", "]") // 'cause snapshots have trailing comas and JSON.parse doesn't like it.
);

test.each(tables.map(name => [name]))("%s table schema", async name => {
  expect(await knex.table(name).columnInfo()).toMatchSnapshot();
});

test("should rollback to nothing", async () => {
  await knex.migrate.rollback();

  const query = `
    SELECT table_name FROM information_schema.tables
    WHERE table_schema = current_schema() AND table_catalog = ?
  `;
  const results = await knex.raw(query, [knex.client.database()]);
  expect(results.rows.map(row => row.table_name).sort()).toMatchInlineSnapshot(`
Array [
  "knex_migrations",
  "knex_migrations_lock",
]
`);
});
