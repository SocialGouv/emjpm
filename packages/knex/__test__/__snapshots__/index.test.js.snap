// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`city table schema 1`] = `
Object {
  "code_postal": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": false,
    "type": "character varying",
  },
  "id": Object {
    "defaultValue": "nextval('city_id_seq'::regclass)",
    "maxLength": null,
    "nullable": false,
    "type": "integer",
  },
  "latitude": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": false,
    "type": "character varying",
  },
  "longitude": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": false,
    "type": "character varying",
  },
  "ville": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": false,
    "type": "character varying",
  },
}
`;

exports[`commentaires table schema 1`] = `
Object {
  "antenne_id": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
  "comment": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "text",
  },
  "created_at": Object {
    "defaultValue": "CURRENT_TIMESTAMP",
    "maxLength": null,
    "nullable": true,
    "type": "timestamp with time zone",
  },
  "id": Object {
    "defaultValue": "nextval('commentaires_co_id_seq'::regclass)",
    "maxLength": null,
    "nullable": false,
    "type": "integer",
  },
  "mandataire_id": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
  "service_id": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
  "ti_id": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
}
`;

exports[`departements table schema 1`] = `
Object {
  "code": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": false,
    "type": "character varying",
  },
  "id": Object {
    "defaultValue": "nextval('departements_id_seq'::regclass)",
    "maxLength": null,
    "nullable": false,
    "type": "integer",
  },
  "id_region": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
  "nom": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": false,
    "type": "character varying",
  },
}
`;

exports[`direction table schema 1`] = `
Object {
  "department_id": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
  "id": Object {
    "defaultValue": "nextval('direction_id_seq'::regclass)",
    "maxLength": null,
    "nullable": false,
    "type": "integer",
  },
  "region_id": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
  "user_id": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
}
`;

exports[`etablissements table schema 1`] = `
Object {
  "adresse": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "code_postal": Object {
    "defaultValue": null,
    "maxLength": 10,
    "nullable": true,
    "type": "character varying",
  },
  "fax": Object {
    "defaultValue": null,
    "maxLength": 15,
    "nullable": true,
    "type": "character varying",
  },
  "id": Object {
    "defaultValue": "nextval('etablissements_id_seq'::regclass)",
    "maxLength": null,
    "nullable": false,
    "type": "integer",
  },
  "id_finess": Object {
    "defaultValue": null,
    "maxLength": 20,
    "nullable": true,
    "type": "character varying",
  },
  "latitude": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "real",
  },
  "longitude": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "real",
  },
  "nom": Object {
    "defaultValue": null,
    "maxLength": 100,
    "nullable": false,
    "type": "character varying",
  },
  "tel": Object {
    "defaultValue": null,
    "maxLength": 15,
    "nullable": true,
    "type": "character varying",
  },
  "ville": Object {
    "defaultValue": null,
    "maxLength": 100,
    "nullable": true,
    "type": "character varying",
  },
}
`;

exports[`geolocalisation_code_postal table schema 1`] = `
Object {
  "cities": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "code_postal": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "id": Object {
    "defaultValue": "nextval('\\"codePostalLatLngs_id_seq\\"'::regclass)",
    "maxLength": null,
    "nullable": false,
    "type": "integer",
  },
  "insee": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "latitude": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "real",
  },
  "longitude": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "real",
  },
}
`;

exports[`knex_migrations_v2 table schema 1`] = `
Object {
  "batch": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
  "id": Object {
    "defaultValue": "nextval('knex_migrations_v2_id_seq'::regclass)",
    "maxLength": null,
    "nullable": false,
    "type": "integer",
  },
  "migration_time": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "timestamp with time zone",
  },
  "name": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
}
`;

exports[`knex_migrations_v2_lock table schema 1`] = `
Object {
  "index": Object {
    "defaultValue": "nextval('knex_migrations_v2_lock_index_seq'::regclass)",
    "maxLength": null,
    "nullable": false,
    "type": "integer",
  },
  "is_locked": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
}
`;

exports[`logs_data table schema 1`] = `
Object {
  "action": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "created_at": Object {
    "defaultValue": "CURRENT_TIMESTAMP",
    "maxLength": null,
    "nullable": true,
    "type": "timestamp with time zone",
  },
  "id": Object {
    "defaultValue": "nextval('logs_data_id_seq'::regclass)",
    "maxLength": null,
    "nullable": false,
    "type": "integer",
  },
  "result": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "user_id": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
}
`;

exports[`magistrat table schema 1`] = `
Object {
  "id": Object {
    "defaultValue": "nextval('magistrat_id_seq'::regclass)",
    "maxLength": null,
    "nullable": false,
    "type": "integer",
  },
  "ti_id": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": false,
    "type": "integer",
  },
  "user_id": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": false,
    "type": "integer",
  },
}
`;

exports[`mandataire_etablissements table schema 1`] = `
Object {
  "created_at": Object {
    "defaultValue": "CURRENT_TIMESTAMP",
    "maxLength": null,
    "nullable": false,
    "type": "timestamp with time zone",
  },
  "etablissement_id": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
  "id": Object {
    "defaultValue": "nextval('\\"mandatairesEtablissements_id_seq\\"'::regclass)",
    "maxLength": null,
    "nullable": false,
    "type": "integer",
  },
  "mandataire_id": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
}
`;

exports[`mandataires table schema 1`] = `
Object {
  "adresse": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": false,
    "type": "character varying",
  },
  "code_postal": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": false,
    "type": "character varying",
  },
  "created_at": Object {
    "defaultValue": "CURRENT_TIMESTAMP",
    "maxLength": null,
    "nullable": true,
    "type": "timestamp with time zone",
  },
  "date_mesure_update": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "timestamp with time zone",
  },
  "department_id": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": false,
    "type": "integer",
  },
  "dispo_max": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
  "email_send": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "timestamp with time zone",
  },
  "etablissement": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": false,
    "type": "character varying",
  },
  "genre": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "id": Object {
    "defaultValue": "nextval('mandataires_id_seq'::regclass)",
    "maxLength": null,
    "nullable": false,
    "type": "integer",
  },
  "latitude": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "real",
  },
  "longitude": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "real",
  },
  "mesures_en_attente": Object {
    "defaultValue": "0",
    "maxLength": null,
    "nullable": false,
    "type": "integer",
  },
  "mesures_en_cours": Object {
    "defaultValue": "0",
    "maxLength": null,
    "nullable": false,
    "type": "integer",
  },
  "nb_secretariat": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "real",
  },
  "secretariat": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "boolean",
  },
  "siret": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "telephone": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": false,
    "type": "character varying",
  },
  "telephone_portable": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "user_id": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
  "ville": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": false,
    "type": "character varying",
  },
  "zip": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "text",
  },
}
`;

exports[`mesures table schema 1`] = `
Object {
  "annee": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "antenne_id": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
  "cabinet": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "civilite": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "code_postal": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "created_at": Object {
    "defaultValue": "CURRENT_TIMESTAMP",
    "maxLength": null,
    "nullable": true,
    "type": "timestamp with time zone",
  },
  "date_ouverture": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "date",
  },
  "department_id": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
  "etablissement": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "etablissement_id": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
  "extinction": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "date",
  },
  "id": Object {
    "defaultValue": "nextval('mesures_id_seq'::regclass)",
    "maxLength": null,
    "nullable": false,
    "type": "integer",
  },
  "is_urgent": Object {
    "defaultValue": "false",
    "maxLength": null,
    "nullable": true,
    "type": "boolean",
  },
  "judgment_date": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "date",
  },
  "latitude": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "real",
  },
  "longitude": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "real",
  },
  "mandataire_id": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
  "numero_dossier": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "numero_rg": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "reason_extinction": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "residence": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "service_id": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
  "status": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "ti_id": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
  "type": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "ville": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
}
`;

exports[`mesures_import table schema 1`] = `
Object {
  "antenne_id": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
  "content": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "jsonb",
  },
  "created_at": Object {
    "defaultValue": "CURRENT_TIMESTAMP",
    "maxLength": null,
    "nullable": false,
    "type": "timestamp with time zone",
  },
  "file_name": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": false,
    "type": "character varying",
  },
  "file_size": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": false,
    "type": "integer",
  },
  "file_type": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": false,
    "type": "character varying",
  },
  "id": Object {
    "defaultValue": "nextval('mesures_import_id_seq'::regclass)",
    "maxLength": null,
    "nullable": false,
    "type": "integer",
  },
  "processed_at": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "timestamp with time zone",
  },
  "service_id": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
  "status": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": false,
    "type": "character varying",
  },
  "user_id": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
}
`;

exports[`regions table schema 1`] = `
Object {
  "id": Object {
    "defaultValue": "nextval('regions_id_seq'::regclass)",
    "maxLength": null,
    "nullable": false,
    "type": "integer",
  },
  "nom": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": false,
    "type": "character varying",
  },
}
`;

exports[`role table schema 1`] = `
Object {
  "id": Object {
    "defaultValue": "nextval('role_id_seq'::regclass)",
    "maxLength": null,
    "nullable": false,
    "type": "integer",
  },
  "name": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": false,
    "type": "character varying",
  },
}
`;

exports[`service_admin table schema 1`] = `
Object {
  "id": Object {
    "defaultValue": "nextval('service_admin_id_seq'::regclass)",
    "maxLength": null,
    "nullable": false,
    "type": "integer",
  },
  "service_id": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
  "user_id": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
}
`;

exports[`service_antenne table schema 1`] = `
Object {
  "address": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "address_city": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "address_street": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "address_zip_code": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "bak_mandataire_id": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
  "contact_email": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "contact_firstname": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "contact_lastname": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "contact_phone": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "created_at": Object {
    "defaultValue": "CURRENT_TIMESTAMP",
    "maxLength": null,
    "nullable": false,
    "type": "timestamp with time zone",
  },
  "date_mesure_update": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "timestamp with time zone",
  },
  "id": Object {
    "defaultValue": "nextval('service_antenne_id_seq'::regclass)",
    "maxLength": null,
    "nullable": false,
    "type": "integer",
  },
  "latitude": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "real",
  },
  "longitude": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "real",
  },
  "mesures_awaiting": Object {
    "defaultValue": "0",
    "maxLength": null,
    "nullable": false,
    "type": "integer",
  },
  "mesures_in_progress": Object {
    "defaultValue": "0",
    "maxLength": null,
    "nullable": false,
    "type": "integer",
  },
  "mesures_max": Object {
    "defaultValue": "0",
    "maxLength": null,
    "nullable": false,
    "type": "integer",
  },
  "name": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "service_id": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": false,
    "type": "integer",
  },
}
`;

exports[`service_tis table schema 1`] = `
Object {
  "id": Object {
    "defaultValue": "nextval('service_tis_id_seq'::regclass)",
    "maxLength": null,
    "nullable": false,
    "type": "integer",
  },
  "service_id": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": false,
    "type": "integer",
  },
  "ti_id": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
}
`;

exports[`services table schema 1`] = `
Object {
  "adresse": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "code_postal": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "created_at": Object {
    "defaultValue": "CURRENT_TIMESTAMP",
    "maxLength": null,
    "nullable": true,
    "type": "timestamp with time zone",
  },
  "department_id": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": false,
    "type": "integer",
  },
  "dispo_max": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
  "email": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "etablissement": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "id": Object {
    "defaultValue": "nextval('services_id_seq'::regclass)",
    "maxLength": null,
    "nullable": false,
    "type": "integer",
  },
  "information": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "latitude": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "real",
  },
  "longitude": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "real",
  },
  "mesures_awaiting": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
  "mesures_in_progress": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
  "nom": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "prenom": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "telephone": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "ville": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
}
`;

exports[`should list all tables 1`] = `
Array [
  "city",
  "commentaires",
  "departements",
  "direction",
  "etablissements",
  "geolocalisation_code_postal",
  "knex_migrations_v2",
  "knex_migrations_v2_lock",
  "logs_data",
  "magistrat",
  "mandataire_etablissements",
  "mandataires",
  "mesures",
  "mesures_import",
  "regions",
  "role",
  "service_admin",
  "service_antenne",
  "service_tis",
  "services",
  "tis",
  "user_antenne",
  "user_role",
  "user_tis",
  "users",
  "view_department_availability",
  "view_mesure_gestionnaire",
  "view_mesure_gestionnaire_tis",
]
`;

exports[`tis table schema 1`] = `
Object {
  "code_postal": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": false,
    "type": "character varying",
  },
  "created_at": Object {
    "defaultValue": "CURRENT_TIMESTAMP",
    "maxLength": null,
    "nullable": false,
    "type": "timestamp with time zone",
  },
  "email": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "etablissement": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": false,
    "type": "character varying",
  },
  "id": Object {
    "defaultValue": "nextval('tis_id_seq'::regclass)",
    "maxLength": null,
    "nullable": false,
    "type": "integer",
  },
  "siret": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "telephone": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "ville": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": false,
    "type": "character varying",
  },
}
`;

exports[`user_antenne table schema 1`] = `
Object {
  "antenne_id": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
  "id": Object {
    "defaultValue": "nextval('user_antenne_id_seq'::regclass)",
    "maxLength": null,
    "nullable": false,
    "type": "integer",
  },
  "user_id": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
}
`;

exports[`user_role table schema 1`] = `
Object {
  "id": Object {
    "defaultValue": "nextval('user_role_id_seq'::regclass)",
    "maxLength": null,
    "nullable": false,
    "type": "integer",
  },
  "role_id": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
  "user_id": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
}
`;

exports[`user_tis table schema 1`] = `
Object {
  "created_at": Object {
    "defaultValue": "CURRENT_TIMESTAMP",
    "maxLength": null,
    "nullable": true,
    "type": "timestamp with time zone",
  },
  "id": Object {
    "defaultValue": "nextval('users_tis_id_seq'::regclass)",
    "maxLength": null,
    "nullable": false,
    "type": "integer",
  },
  "ti_id": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
  "user_id": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
}
`;

exports[`users table schema 1`] = `
Object {
  "active": Object {
    "defaultValue": "false",
    "maxLength": null,
    "nullable": true,
    "type": "boolean",
  },
  "cabinet": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "created_at": Object {
    "defaultValue": "CURRENT_TIMESTAMP",
    "maxLength": null,
    "nullable": false,
    "type": "timestamp with time zone",
  },
  "email": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "id": Object {
    "defaultValue": "nextval('users_id_seq'::regclass)",
    "maxLength": null,
    "nullable": false,
    "type": "integer",
  },
  "last_login": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "timestamp with time zone",
  },
  "nom": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "password": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": false,
    "type": "character varying",
  },
  "prenom": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "reset_password_expires": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "timestamp with time zone",
  },
  "reset_password_token": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "type": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "username": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": false,
    "type": "character varying",
  },
}
`;

exports[`view_department_availability table schema 1`] = `
Object {
  "department_id": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
  "mesures_awaiting": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "bigint",
  },
  "mesures_in_progress": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "bigint",
  },
  "mesures_max": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "bigint",
  },
}
`;

exports[`view_mesure_gestionnaire table schema 1`] = `
Object {
  "dep_nom": Object {
    "defaultValue": null,
    "maxLength": 255,
    "nullable": true,
    "type": "character varying",
  },
  "department_id": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
  "discriminator": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "text",
  },
  "id": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "text",
  },
  "mandataire_id": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
  "mesures_awaiting": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
  "mesures_in_progress": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
  "mesures_max": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
  "nom": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "character varying",
  },
  "remaining_capacity": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
  "service_id": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
}
`;

exports[`view_mesure_gestionnaire_tis table schema 1`] = `
Object {
  "discriminator": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "text",
  },
  "id": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "text",
  },
  "mandataire_id": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
  "service_id": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
  "ti_id": Object {
    "defaultValue": null,
    "maxLength": null,
    "nullable": true,
    "type": "integer",
  },
}
`;
